server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/DataArchiveTask.java

```diff
@@ -76,8 +63,17 @@ protected boolean doStart(JobContext context) throws Exception {
                 log.info("Job is terminated,jobIdentity={}", context.getJobIdentity());
                 break;
             }
-            if (parameters.getJobType() == JobType.MIGRATE) {
-                syncTable(tableIndex, parameters);
+            if (parameters.getJobType() == JobType.MIGRATE && parameters.isTableStructureSyncEnabled()) {
+                try {
+                    DLMTableStructureSynchronizer.sync(
```

Comment:
The `DLMTableStructureSynchronizer.sync()` will not throw any exception if failed to execute sync SQL script. You may need refine the code logic.